src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
> Validating types may be outside of the purpose of php_phongo_manager_prep_authmechanismproperties() and php_phongo_manager_prep_uri_options()

so we will not need to change it's type from void to another type , as we will implement a new validation function that applied on BSON values instead of zval - if I got what you mean - and since we are converting the zval to bson inside the [`phongo_manager_init`](https://github.com/mongodb/mongo-php-driver/blob/f8d0a3301d68ace9261db9d19ab9d4fcdf724984/php_phongo.c#L1784) function which is invoked after `php_phongo_manager_prep_uri_options` function but in this case we will not stop proceeding the trying to initialize the object.
> so that we don't proceed with trying to initialize the object any further.